receiver/receivercreator/config.go

```diff
@@ -90,8 +90,10 @@ type Config struct {
 }
 
 type DiscoveryConfig struct {
-	Enabled         bool     `mapstructure:"enabled"`
-	IgnoreReceivers []string `mapstructure:"ignore_receivers"`
+	Enabled              bool     `mapstructure:"enabled"`
+	IgnoreReceivers      []string `mapstructure:"ignore_receivers"`
+	DefaultLogsDiscovery bool     `mapstructure:"default_logs_discovery"`
```

Comment:
Does this field enable logs while "enabled" enables metrics? Looks pretty inconsistent. Should we rename them?

Reply:
`enabled` setting enables the discovery feature in general.

`default_logs_discovery` provides the option to collect logs from all discovered containers by default. If this is not configured/enabled then only annotated Pods will have their logs collected. It's mostly a convenience setting for users that want to collect all logs by default with only a few of them requiring special handling through the annotations.

I don't have very strong preference for supporting this initially so if you think we should keep it out I'm happy to remove it. Let me know what you think!

Reply:
This is super confusing to me as a user. Why don't we have this option for logs only and not for metrics?

Reply:
I think we need two orthogonal configuration options eventually:
- _enabled by annotation_ vs _enabled always and disabled by annotations_
- _metrics_ vs _logs_

Reply:
I think we should handle that separately, in another PR

Reply:
Sounds good. I have removed that now. I will file a follow up issue to keep track of it, once this PR is merged.